optimize/backend/src/main/java/io/camunda/optimize/rest/DefinitionRestService.java

```diff
@@ -212,43 +197,34 @@ public List<TenantWithDefinitionsResponseDto> getDefinitionsGroupedByTenant(
    * @param type The type of the definition (process or decision).
    * @return the definition xml requested or json error structure on failure
    */
-  @GET
-  @Produces(value = {MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
-  @Path("/{type}/xml")
+  @GetMapping(path = "/{type}/xml", produces = MediaType.APPLICATION_XML_VALUE)
   @CacheRequest
-  public Response getDefinitionXml(
-      @Context final ContainerRequestContext requestContext,
-      @PathParam("type") final DefinitionType type,
-      @QueryParam("key") final String key,
-      @QueryParam("version") final String version,
-      @QueryParam("tenantId") final String tenantId) {
-    final String userId = sessionService.getRequestUserOrFailNotAuthorized(requestContext);
+  public ResponseEntity<String> getDefinitionXml(
+      @PathVariable("type") final DefinitionType type,
+      @RequestParam(name = "key", required = false) final String key,
+      @RequestParam(name = "version", required = false) final String version,
+      @RequestParam(name = "tenantId", required = false) final String tenantId,
+      final HttpServletRequest request) {
+    final String userId = sessionService.getRequestUserOrFailNotAuthorized(request);
     final Optional<DefinitionOptimizeResponseDto> definitionDto =
         definitionService.getDefinitionWithXml(type, userId, key, version, tenantId);
 
     if (definitionDto.isEmpty()) {
       logAndThrowNotFoundException(type, key, version);
     }
-    final Response.ResponseBuilder responseBuilder = Response.ok().type(MediaType.APPLICATION_XML);
+
+    final BodyBuilder bodyBuilder = ResponseEntity.ok().contentType(MediaType.APPLICATION_XML);
     if (isDefinitionVersionSetToAllOrLatest(version)) {
-      addNoStoreCacheHeader(responseBuilder);
-    }
-    switch (type) {
-      case PROCESS:
-        final ProcessDefinitionOptimizeDto processDef =
-            (ProcessDefinitionOptimizeDto) definitionDto.get();
-        responseBuilder.entity(processDef.getBpmn20Xml());
-        break;
-      case DECISION:
-        final DecisionDefinitionOptimizeDto decisionDef =
-            (DecisionDefinitionOptimizeDto) definitionDto.get();
-        responseBuilder.entity(decisionDef.getDmn10Xml());
-        break;
-      default:
-        throw new BadRequestException("Unknown DefinitionType:" + type);
+      addNoStoreCacheHeader(bodyBuilder);
     }
 
-    return responseBuilder.build();
+    final String xml =
+        switch (type) {
```

Comment:
Does't a switch without default branch go against the checkstyle rules?

Reply:
A default here isn't necessary because the enum only has two values, and both are handled here. A default case would never be executed